### PR TITLE
fix: various fetch issues

### PR DESCRIPTION
**Pull request by ericlewis**
This tries to bring fetch more in line with specs & fixes some bugs

### CODE REVIEW

**Comment by theolampert** [comment]
Thanks for this @ericlewis I'll take a look

File: Sources/ECMASwift/API/Headers/Headers.swift [hunk 1]
```diff
@@ -41,8 +41,12 @@ final class Headers: NSObject, HeadersExports {
 /// Helper to register the ``Headers`` API with a context.
 struct HeadersAPI {
     func registerAPIInto(context: JSContext) {
-        let headersClass: @convention(block) () -> Headers = {
-            Headers()
+        let headersClass: @convention(block) (JSValue?) -> Headers = { headers in
+            if let headers {
```
Comment (theolampert) [inline Sources/ECMASwift/API/Headers/Headers.swift:45]: I think you'll also need to check if the underlying `JSValue` is also `null` or `undefined` here, there's a helper for this. Something like:

```swift
if let headers, headers.hasValue {
```

**Review by theolampert — COMMENTED** [review]
@ericlewis Looks good thank you, I only had one comment.

Would you also be able to let me know what issues you ran into so I can add those to the header and fetch tests?

File: Sources/ECMASwift/API/Headers/Headers.swift (hunk 1 above)
Comment (ericlewis) [reply Sources/ECMASwift/API/Headers/Headers.swift:45]: sorry for not seeing this. did you make the change?

Comment (theolampert) [reply Sources/ECMASwift/API/Headers/Headers.swift:45]: Yep, all good, thanks again for the contribution!

**Pull request merged**